src/main/java/io/github/nucleuspowered/nucleus/modules/playerinfo/commands/NearCommand.java

```diff
@@ -0,0 +1,125 @@
+        pi.put("maxexempt", PermissionInformation.getWithTranslation("permission.near.maxexempt", SuggestedLevel.MOD));
+        return pi;
+    }
+
+    @Override
+    public CommandElement[] getArguments() {
+        return new CommandElement[] {
+                GenericArguments.optionalWeak(requirePermissionArg(GenericArguments.user(Text.of(this.playerKey)), this.permissions.getOthers())),
+                GenericArguments.optionalWeak(GenericArguments.integer(Text.of(this.radiusKey)))
+        };
+    }
+
+    //near radius
+    //near player radius
+    @Override
+    protected CommandResult executeCommand(CommandSource src, CommandContext args, Cause cause) throws Exception {
+        final User user = this.getUserFromArgs(User.class, src, this.playerKey, args);
+        final Location<World> location;
+        if (user.isOnline()) {
+            location = user.getPlayer().get().getLocation();
+        } else {
+            Supplier<ReturnMessageException> returnMessageExceptionSupplier = () -> ReturnMessageException.fromKey("command.near.location.nolocation", user.getName());
+            location = Nucleus.getNucleus().getUserDataManager().get(user)
+                    .orElseThrow(returnMessageExceptionSupplier)
+                    .get(CoreUserDataModule.class).getLogoutLocation()
+                    .orElseThrow(returnMessageExceptionSupplier);
+        }
+
+        int radius = maxRadius;
+        Optional<Integer> radiusOpt = args.getOne(radiusKey);
+        if(radiusOpt.isPresent()) {
+            int inputRadius = radiusOpt.get();
+            //Check if executer has max radius override permission
+            if (inputRadius > maxRadius && this.permissions.testSuffix(src, "maxexempt")) {
+                radius = inputRadius;
+            } else {
+                src.sendMessage(Nucleus.getNucleus().getMessageProvider().getTextMessageWithFormat("command.near.radiustoobig", this.maxRadius));
+            }
+        }
+
+        List<Player> nearbyPlayers = location.getExtent().getNearbyEntities(location.getPosition(), radius).stream()
+                .filter(Player.class::isInstance)
+                .map(Player.class::cast)
+                .filter(e -> e.getUniqueId() != user.getUniqueId())
+                .collect(Collectors.toList());
+
+        Text.Builder tb = Nucleus.getNucleus().getMessageProvider()
+                .getTextMessageWithFormat(
+                        "command.near.playersnear",
+                        user.getName()
+                ).toBuilder();
+        src.sendMessage(tb.build());
+
+
+        List<Text> messages = new ArrayList<>(nearbyPlayers.size());
+        nearbyPlayers.forEach(nearbyPlayer -> {
+            Text.Builder line = Text.builder();
+            line.append(Text.of(nearbyPlayer.getName(), " ", Math.abs(nearbyPlayer.getLocation().getPosition().distance(location.getPosition())), "M"))
```

Comment:
Why M? At least make it lower case (I know a block is meant to be 1m side length). Better as blocks, though?

This should really be in a translatable string that has these as tokens.